extensions/sinks/influx2/influx2.go

```diff
@@ -111,8 +114,8 @@ func (m *influxSink2) Open(ctx api.StreamContext) (err error) {
 
 func (m *influxSink2) Collect(ctx api.StreamContext, data interface{}) error {
 	logger := ctx.GetLogger()
-	if m.hasTransform {
-		jsonBytes, _, err := ctx.TransformOutput(data)
+	if m.hasTransform || len(m.fields) > 0 {
```

Comment:
can keep the old logic, only deal with dateTemplate. Make the selectMap an util function to be used to handle fields